internal/controller/etcd/reconcile_delete.go

```diff
@@ -0,0 +1,116 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package etcd
+
+import (
+	"fmt"
+	"time"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/internal/common"
+	ctrlutils "github.com/gardener/etcd-druid/internal/controller/utils"
+	"github.com/gardener/etcd-druid/internal/operator/component"
+	"github.com/gardener/etcd-druid/internal/utils"
+	"github.com/gardener/gardener/pkg/controllerutils"
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// triggerDeletionFlow is the entry point for the deletion flow triggered for an etcd component which has a DeletionTimeStamp set on it.
+func (r *Reconciler) triggerDeletionFlow(ctx component.OperatorContext, logger logr.Logger, etcdObjectKey client.ObjectKey) ctrlutils.ReconcileStepResult {
+	deleteStepFns := []reconcileFn{
+		r.recordDeletionStartOperation,
+		r.deleteEtcdResources,
+		r.verifyNoResourcesAwaitCleanUp,
+		r.removeFinalizer,
+	}
+	for _, fn := range deleteStepFns {
+		if stepResult := fn(ctx, etcdObjectKey); ctrlutils.ShortCircuitReconcileFlow(stepResult) {
+			return r.recordIncompleteDeletionOperation(ctx, logger, etcdObjectKey, stepResult)
+		}
+	}
+	return ctrlutils.DoNotRequeue()
+}
+
+func (r *Reconciler) deleteEtcdResources(ctx component.OperatorContext, etcdObjKey client.ObjectKey) ctrlutils.ReconcileStepResult {
+	etcd := &druidv1alpha1.Etcd{}
+	if result := r.getLatestEtcd(ctx, etcdObjKey, etcd); ctrlutils.ShortCircuitReconcileFlow(result) {
+		return result
+	}
+	operators := r.operatorRegistry.AllOperators()
+	deleteTasks := make([]utils.OperatorTask, 0, len(operators))
+	for kind, operator := range operators {
+		operator := operator
```

Comment:
Sure this was already on our list but an explicit TODO would help us remember.

Reply:
@renormalize thanks for pointing it out. Could you may be at a note in https://github.com/gardener/etcd-druid/pull/789/ about this, so that we don't forget?